build.sbt

```diff
@@ -75,6 +75,8 @@ ThisBuild / publishTo := {
 }
 ThisBuild / pomIncludeRepository := (_ => false) // drop repos other than Maven Central from POM
 ThisBuild / mimaPreviousArtifacts := Set.empty
+// limit the number of concurrent test so testQuick works
+Global / concurrentRestrictions += Tags.limit(Tags.Test, 4)
```

Comment:
What's going on here?  I never use `testQuick`.

Reply:
When I run `~testQuick` at the root level, I noticed that it's running 8 tests at a time, each vying for memory and CPU, and sometimes it would get stuck like it's deadlocked. It could also be plain memory leak somewhere. Project matrix makes it easy to have parallel subprojects. I think global 4 is a sweet spot for 2G of RAM I'm giving to this build.